subscription-sender/internal/queue/notification.go

```diff
@@ -0,0 +1,31 @@
+package queue
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/RinaDish/subscription-sender/internal/services"
+	"github.com/nats-io/nats.go"
+)
+
+func (queue *Queue) Subscribe(ctx context.Context) error {
+	_, err := queue.QueueConn.Subscribe(queue.subscriptionTopicName, func(msg *nats.Msg) {
+        queue.getMessage(msg, ctx)
+    })
+
+	if err != nil {
+        return err
+    }
+
+	queue.logger.Infof("Listening for messages on 'subscription' topic...")
+
+	return nil
+}
+
+func  (queue *Queue) getMessage(msg *nats.Msg, ctx context.Context) {
+	var unmrshBody services.Notification
+	_ = json.Unmarshal(msg.Data, &unmrshBody)
```

Comment:
What if message would not be unmarshalled? 
It's quite critical part of subscription-sender, because broker messaging is main way of communication with it, so it is bad place to ignore errors

Reply:
done